numpy/lib/tests/test_arraysetops.py

```diff
@@ -247,6 +247,12 @@ def test_union1d(self):
         c = union1d(a, b)
         assert_array_equal(c, ec)
 
+        x = np.array([[0, 1, 2], [3, 4, 5]])
+        y = np.array([0, 1, 2, 3, 4])
+        ez = np.array([0, 1, 2, 3, 4, 5])
+        z = union1d(x, y)
+        assert_array_equal(z, ez)
```

Comment:
A comment above this block linking to the issue with `gh-10340`, and a remark about flattening, might be nice.

Reply:
Comment added.